hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java

```diff
@@ -101,6 +102,16 @@ public ContainerManagerImpl(
       final PipelineManager pipelineManager,
       final Table<ContainerID, ContainerInfo> containerStore)
       throws IOException {
+    this(conf, scmHaManager, pipelineManager, containerStore, null);
```

Comment:
Could we avoid using this Ctor ? We may need to change the related test cases.

Reply:
Same https://issues.apache.org/jira/browse/HDDS-4634.

This is because MockHAManager does not have a mock buffer implementation.

Reply:
Could we implement the mock buffer in this PR ? Since missing this change will pollute the production code, and will also give a burden to the on-going PR of deleted block log.